blackboxopt/utils.py

```diff
@@ -80,3 +82,20 @@ def filter_pareto_efficient(
     pareto_efficient_mask = mask_pareto_efficient(losses)
 
     return list(compress(evaluations, pareto_efficient_mask))
+
+
+def sort_evaluations(evaluations: List[Evaluation]) -> List[Evaluation]:
+    return sorted(
+        evaluations,
+        key=lambda e: hashlib.md5(
+            pickle.dumps(
+                [
+                    sorted(e.configuration.items()),
+                    sorted(e.objectives.items()),
+                    sorted(e.settings.items()),
+                    sorted(e.constraints.items()) if e.constraints else {},
+                    sorted(e.context.items()) if e.context else {},
+                ]
+            )
+        ).hexdigest(),
+    )
```

Comment:
Thanks a lot for evaluating different alternatives, I'd invite you to add a small ADR to document that reasoning.

Reply:
Done